e2e_playwright/cached_widget_replay_widget_key.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import streamlit as st
+
+st.button("click to rerun")
+
+side_effects = []
+
+
+@st.experimental_memo(experimental_allow_widgets=True)
```

Comment:
nit: I think we should use `cache_data` here instead which also supports `experimental_allow_widgets`.